web/modules/custom/rdf_etl/tests/src/Kernel/ConvertToAdms2Test.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\rdf_etl\Kernel;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\rdf_etl\Plugin\EtlAdms2ConvertPassInterface;
+use Drupal\Tests\rdf_entity\Traits\RdfDatabaseConnectionTrait;
+use EasyRdf\Graph;
+use EasyRdf\GraphStore;
+
+/**
+ * Tests the 'convert_to_adms2' process step plugin.
+ *
+ * @group rdf_etl
+ */
+class ConvertToAdms2Test extends KernelTestBase {
+
+  use RdfDatabaseConnectionTrait;
+
+  /**
+   * The  ADMS v1 to v2 transformation plugin manager.
+   *
+   * @var \Drupal\rdf_etl\Plugin\EtlAdms2ConvertPassPluginManager
+   */
+  protected $adms2ConverPassPluginManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'rdf_entity',
+    'rdf_etl',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->setUpSparql();
+
+    $options = $this->sparql->getConnectionOptions();
+    $connection_uri = "http://{$options['host']}:{$options['port']}/sparql-graph-crud";
+
+    $this->adms2ConverPassPluginManager = $this->container->get('plugin.manager.etl_adms2_convert_pass');
+    $graph_uri = EtlAdms2ConvertPassInterface::TEST_GRAPH;
+
+    $rdf_data = '';
+    // Collect RDF testing data from plugins.
+    foreach ($this->adms2ConverPassPluginManager->getDefinitions() as $plugin_id => $definition) {
+      /** @var \Drupal\rdf_etl\Plugin\EtlAdms2ConvertPassInterface $plugin */
+      $plugin = $this->adms2ConverPassPluginManager->createInstance($plugin_id);
+      if ($plugin_rdf_data = $plugin->getTestingRdfData()) {
+        $rdf_data .= "$plugin_rdf_data\n";
+      }
+    }
+
+    if ($rdf_data = $this->prepareRdfData($rdf_data)) {
+      $graph_store = new GraphStore($connection_uri);
+      $graph = new Graph();
+      $graph->parse($rdf_data);
+      $out = $graph_store->replace($graph, $graph_uri);
+      if (!$out->isSuccessful()) {
+        throw new \Exception("Cannot import RDF data from plugin '$plugin_id' in graph '$graph_uri'.");
```

Comment:
This `$plugin_id` variable seems to be outside the `foreach` loop so it will take the last value in the array regardless of which plugin's data actually fail, no?